### DISCUSSION TITLE
build.js error for a new package

### DISCUSSION THREAD

**Discussion by Weicai-Ling** [cockpit-project/cockpit#20285] [Q&A] [answered]
I want to build a package with build.js. i copied the package of "pinger" from directory of "examples" to "mydev" under directory of "pkg". it shows errors when I run `build.js mydev`. 

`[wan@easyhpc cockpit]$ ls pkg/mydev`
``manifest.json  pinger.js  ping.html`
`
[wan@easyhpc cockpit]$ ./build.js mydev
18:38:15: Build finished
✘ [ERROR] ENOENT: no such file or directory, open '/home/wan/cockpit/dist/mydev/po.cs.js' [plugin cockpitPoEsbuildPlugin]

    /home/wan/cockpit/node_modules/esbuild/lib/main.js:1496:32:
      1496 │             const value = await callback(result);
           ╵                                 ^

    at Object.openSync (node:fs:590:3)
    at Object.writeFileSync (node:fs:2202:35)
    at file:///home/wan/cockpit/pkg/lib/cockpit-po-plugin.js:99:12
    at new Promise (<anonymous>)
    at buildFile (file:///home/wan/cockpit/pkg/lib/cockpit-po-plugin.js:48:12)
    at run (file:///home/wan/cockpit/pkg/lib/cockpit-po-plugin.js:119:17)
    at file:///home/wan/cockpit/pkg/lib/cockpit-po-plugin.js:131:75
    at /home/wan/cockpit/node_modules/esbuild/lib/main.js:1496:33
    at runOnEndCallbacks (/home/wan/cockpit/node_modules/esbuild/lib/main.js:1528:9)
    at buildResponseToResult (/home/wan/cockpit/node_modules/esbuild/lib/main.js:1057:7)

  This error came from the "onEnd" callback registered here:

    /home/wan/cockpit/node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///home/wan/cockpit/pkg/lib/cockpit-po-plugin.js:131:15)
    at handlePlugins (/home/wan/cockpit/node_modules/esbuild/lib/main.js:1293:21)
`
It also shows the similar error, when I run build.js for my packages. who could please advise how to build packages with build.js ? 

THANKS. 
Forrest Ling

**Comment by martinpitt** [comment, marked as answer]
@Weicai-Ling examples/pinger is *not* meant to be covered by `./build.js` -- it's a small standalone page which just works without any building. Just copy it to ~/.local/share/cockpit/pinger or /usr/local/share/cockpit/pinger.

**Reply by Weicai-Ling** [reply]
@martinpitt , understand, Thanks for reply, very HELPFUL!

**Reply by Weicai-Ling** [reply]
@martinpitt I modified my codes from examples/pinger to a new package by adding some .jsx and .css files, could you please advise any best practices in building these codes into a dist folder for symlink with "ln -s to ~/.local/share/cockpit " ?  
THANKS in advance!
 
Forrest Weicai-Ling

**Reply by martinpitt** [reply]
@Weicai-Ling For a full build chain please have a look at https://cockpit-project.org/blog/cockpit-starter-kit.html . This provides all the scaffolding, and you can just drop in your .jsx and .css files.

**Reply by Weicai-Ling** [reply]
@martinpitt Thanks for your quick reply. I had saw the link before, I thought the package of starter-kit was not the updated version, as it seemed it was not with latest React version. OK, I would like to look into the starter-kit and develop my codes based on the package.  

THANKS👍 

Forrest Weicai-Ling

**Reply by Weicai-Ling** [reply]
I downloaded the package and built it successfully. This is what I want. THANKS!

**Reply by martinpitt** [reply]
Great!
